src/hotspot/share/jvmci/jvmci.cpp

```diff
@@ -56,6 +56,11 @@ const char* JVMCI::_fatal_log_filename = nullptr;
 void jvmci_vmStructs_init() NOT_DEBUG_RETURN;
 
 bool JVMCI::can_initialize_JVMCI() {
```

Comment:
For the non libgraal case is this test really complete?  I guess I'm not convinced that the tests in this method are sufficient the safe init point.  Maybe it would be simpler to have a `JVMCI::_can_be_initialized` flag which is set at the correct point during the init steps in the same way that `CompileBroker::_initialized` is set.  Having the order explicitly spelled out there seems better than trying to infer it from other state.

Reply:
We've been running this code for a long time (Graal unit tests on jargraal, Native Image building of libgraal etc) so I don't see any reason to double-guess its correctness now. I suggest we leave it as is and address problems when/if they show up.

Reply:
Ok.